laalaa/apps/advisers/importer.py

```diff
@@ -140,24 +168,32 @@ def import_offices(self):
             'count': 0}
 
         def office(data):
-            loc = location(join(
+            self.check_interrupt()
+            loc = location((
                 data['Address Line 1'],
                 data['Address Line 2'],
                 data['Address Line 3'],
                 data['City'],
                 data['Postcode']))
-            org = models.Organisation.objects.filter(firm=data['Firm Number'])
-            office, created = models.Office.objects.get_or_create(
-                telephone=data['Telephone Number'],
-                account_number=data['Account Number'].upper(),
-                organisation_id=org[0].id,
-                location=loc)
+            org = models.Organisation.objects.filter(firm=data['Firm Number']).first()
+            account_number = data['Account Number'].upper()
+            try:
+                _office = models.Office.objects.get(account_number=account_number)
+                _office.telephone = data['Telephone Number']
+                _office.organisation_id = org.id
+                _office.location_id = loc.id
+                _office.save()
+            except models.Office.DoesNotExist:
+                models.Office.objects.create(
+                    telephone=data['Telephone Number'],
+                    account_number=account_number,
+                    organisation_id=org.id,
+                    location_id=loc.id)
```

Comment:
I'm assuming this change allows for varying data for the same office by overwriting offices with an already seen `account_number`? In which case, I think you could use `update_or_create`?